src/actions/onboardingActions.js

```diff
@@ -193,6 +255,72 @@ export const setupWalletAction = (enableBiometrics?: boolean) => {
   };
 };
 
+export const walletSetupAction = (enableBiometrics?: boolean) => {
+  return async (dispatch: Dispatch, getState: GetState) => {
+    const {
+      onboarding: {
+        pinCode,
+        wallet: importedWallet, // wallet was already added in import step
+        user: onboardingUsername,
+      },
+    } = getState();
+
+    logBreadcrumb('onboarding', 'walletSetupAction: checking for pinCode');
+    if (!pinCode) {
+      reportLog('walletSetupAction failed: no pinCode');
+      return;
+    }
+
+    const isImported = !!importedWallet;
+
+    logBreadcrumb('onboarding', 'walletSetupAction: creating new mnemonic if importedWallet is not present');
+    // will return new mnemonic if importedWallet is not present
+    const mnemonic = importedWallet?.mnemonic || generateMnemonicPhrase();
+
+    // create wallet object
+    const ethersWallet = ethers.Wallet.fromMnemonic(mnemonic);
+
+    // raw private key will be removed from reducer once registration finishes
+    const { address, privateKey } = ethersWallet;
+
+    dispatch(saveDbAction('wallet', { wallet: { data: { address, privateKey } } }));
+    logBreadcrumb('onboarding', 'walletSetupAction: dispatching SET_WALLET');
+    dispatch({ type: SET_WALLET, payload: { address, privateKey } });
```

Comment:
Can i just check where the `privateKey` is being removed from the store? I'm just paranoid that there's a security risk here because until the value is removed, the `privateKey` could be dumped...

Reply:
It is being removed by dispatching action:- dispatch(removePrivateKeyFromMemoryAction()) which is being called in appNavigation in componentDidUpdate.

Reply:
Thank you!